### PR TITLE
User login

### PR DESCRIPTION
**Pull request by santimc**

### CODE REVIEW

File: pyrobots/src/App.js [hunk 1]
```diff
@@ -43,12 +44,12 @@ function App() {
...
             renders the first one that matches the current URL. */}
         <Routes>
-          <Route path="/" element={<Home/>}/>
-          <Route path="/create_user" element={<Create_user/>}/>
-          <Route path="/login" element={<Login/>}/>
-          <Route path="/list_matches" element={<List_matches/>}/>
-          <Route path="/create_match" element={<Create_match/>}/>
-          <Route path="/create_robot" element={<Create_robot/>}/>
+          <Route path="/" element={<Home />} />
+          <Route path="/create_user" element={<RequireToken><Create_user /></RequireToken>} />
```
Comment (fedevirgolini) [inline pyrobots/src/App.js]: I think that for create_user page token is not required.

File: pyrobots/src/login/Login.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  const handleInputChange = (event) => {
+    setUser({
+      ...user,
+      [event.target.name]: event.target.value
+    })
+  }
+
+  const login = () => {
+    if (user.email === '' && user.password === '') { return } else {
```
Comment (fedevirgolini) [inline pyrobots/src/login/Login.js]: What about asking for an empty email *or* empty password?

File: pyrobots/src/login/Login.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          setErrors(error.message);
+        });
+    }
+
+  }
+  return (
+    <>
+      <h2>Login</h2>
+      {fetchToken() ? (<p>You are logged in!</p>) : (
+        <form>
```
Comment (fedevirgolini) [inline pyrobots/src/login/Login.js]: forms shoud have a `onSubmit=` atribute, don't they?

File: pyrobots/src/login/Login.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          <label>Input Username: </label>
+          <input type='email' name="email" onChange={handleInputChange} />
```
Comment (fedevirgolini) [inline pyrobots/src/login/Login.js:53]: label says "Input Username:" but the input is read as email.
Remmember that for us email != username.

File: pyrobots/src/login/Login.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+          <label>Input Password: </label>
+          <input type='password' name="password" onChange={handleInputChange} />
+          <button type='button' onClick={login}>Login</button>
```
Comment (fedevirgolini) [inline pyrobots/src/login/Login.js]: Maybe `type=` should be `'submit'`, in that case `onClick=` wouldn't be necessary.

File: pyrobots/src/login/Login.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  const [error, setErrors] = useState('')
+
+  const handleInputChange = (event) => {
+    setUser({
+      ...user,
+      [event.target.name]: event.target.value
+    })
+  }
+
+  const login = () => {
```
Comment (fedevirgolini) [inline pyrobots/src/login/Login.js]: According to Santi, this function should be called handleSubmit... But this name really sounds what this function does, so is up to you if you want to change it.

**Review by fedevirgolini — CHANGES_REQUESTED** [review]
I left some comments with some suggestions, but in general everything looks good to me.
I ran the test and had an error. It says: `SyntaxError: Cannot use import statement outside a module`. Do you have any idea how can I resolve it?

File: pyrobots/src/App.js (hunk 1 above)
Comment (santimc) [reply pyrobots/src/App.js]: > `SyntaxError: Cannot use import statement outside a module`

I tried to change the syntax of the axios module import but the error kept appearing. The way I did it is the same as in the axios documentation.

**Comment by santimc** [comment]
@fedevirgolini I made changes where you suggested.

**Review by fedevirgolini — APPROVED** [review]

**Pull request merged**